lib/compile-uri/compile-params.js

```diff
@@ -1,22 +1,51 @@
-module.exports = function compileParams(hrefVariablesElement) {
-  const params = {};
+function getRequired(memberElement) {
+  const typeAttributes = memberElement.attributes.getValue('typeAttributes') || [];
+  return typeAttributes.includes('required');
+}
 
-  if (!hrefVariablesElement) return params;
 
-  hrefVariablesElement.forEach((valueElement, keyElement, memberElement) => {
-    const name = keyElement.toValue();
-    const typeAttributes = memberElement.attributes.getValue('typeAttributes') || [];
-    const values = valueElement.attributes.getValue('enumerations') || [];
+function getDefault(valueElement) {
+  return valueElement ? valueElement.attributes.getValue('default') : undefined;
+}
 
+
+function getExample(valueElement) {
+  if (valueElement) {
     const example = valueElement.toValue();
-    params[name] = {
-      required: Array.from(typeAttributes).includes('required'),
-      default: valueElement.attributes.getValue('default'),
-      example: typeof example === 'undefined' || example === null ? values[0] : example,
-      values,
-    };
-
-    return params;
-  });
-  return params;
-};
+
+    if (typeof example === 'undefined' || example === null) {
+      const values = valueElement.attributes.getValue('enumerations') || [];
```

Comment:
```suggestion
      const values = getValues(valueElement);
```

Should this use `getValues`?

Reply:
It could. I wasn't sure whether it's worth the dependency between functions.